### PR TITLE
Admision Controller for Insights

### PR DESCRIPTION
**Pull request by baderbuddy**
**Why This PR?**
This adds an Admission Controller which will apply policies defined in Fairwinds Insights to the user's cluster.

Fixes #

**Changes**
Changes proposed in this pull request:

*
*

**Checklist:**

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or helm-docs has been run for the charts that support it.

### CODE REVIEW

File: stable/insights-admission/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+| nameOverride | string | `""` |  |
+| nodeSelector | object | `{}` |  |
+| podAnnotations | object | `{}` |  |
+| podSecurityContext | object | `{}` |  |
+| replicaCount | int | `1` |  |
+| resources.limits.cpu | int | `1` |  |
+| resources.limits.memory | string | `"2Gi"` |  |
+| resources.requests.cpu | string | `"100m"` |  |
+| resources.requests.memory | string | `"128Mi"` |  |
+| rules[0].apiGroups[0] | string | `"apps"` |  |
```
Comment (rbren) [inline stable/insights-admission/README.md]: If you add some comments to `values.yaml`, it'll clean this up a bit. Goldilocks is a great example.

Comment (baderbuddy) [reply stable/insights-admission/README.md]: Thanks!

**Comment by sudermanjr** [comment]
Looks like you're relying on cert-manager for certificate generation. You'll probably want to setup an e2e pre-script for this that installs cert-manager for your tests.

incubator/letsencrypt-setup is a good example

**Comment by baderbuddy** [comment]
Oh actually this is using the Kubernetes CSR. I'll have to take a look at it and figure out where the v1 is coming from. It looks like in the code it's explicitly using v1beta1.

File: stable/insights-admission/templates/certificate-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+kind: ClusterRole
+metadata:
+  name: {{ include "insights-admission.fullname" . }}-webhook
+  labels:
+    {{- include "insights-admission.labels" . | nindent 4 }}
+rules:
+  - apiGroups:
+      - ''
+    resources:
+      - 'secrets'
```
Comment (rbren) [inline stable/insights-admission/templates/certificate-rbac.yaml]: why do we need cluster-wide secrets?

**Comment by baderbuddy** [comment]
Going back to draft until I can get the tests fixed.

**Comment by baderbuddy** [comment]
I removed the job to update the certs and am relying on cert-manager now. That should reduce the complexity a lot. Except it looks like we still needed a CA, so there's a little cert-manager complexity there.

File: stable/insights-admission/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v2
+name: insights-admission
+description: A Validating Webhook Admission Controller that utilizes Fairwinds Insights.
+type: application
+version: 0.1.0
+appVersion: 0.1.0
```
Comment (rbren) [inline stable/insights-admission/Chart.yaml]: should this be the current version of Insights?

**Review by rbren — APPROVED** [review]

File: stable/tls.crt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+-----BEGIN CERTIFICATE-----
```
Comment (rbren) [inline stable/tls.crt]: is this supposed to be here?

Comment (baderbuddy) [reply stable/tls.crt]: No

Comment (baderbuddy) [reply stable/tls.crt]: Fixed it!

**Review by sudermanjr — CHANGES_REQUESTED** [review]
Overall looks great. Some minor documentation and questions inline.

Love to see that `helm create`, `helm-docs`, and the charts api v2 are being used!

File: stable/insights-admission/values.yaml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  organization: ""
+  cluster: ""
+  host: https://insights.fairwinds.com
+  base64token: ""
+
+image:
+  repository: quay.io/fairwinds/insights-admission-controller
+  pullPolicy: Always
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "0.1"
```
[the remark this answers is not in the document]
Comment (baderbuddy) [reply stable/insights-admission/values.yaml]: The default value I had (accidentally) already not been using. I cleaned up the comment.

[the remark this answers is not in the document]
Comment (baderbuddy) [reply stable/insights-admission/values.yaml]: Reversed course, AppVersion is the default tag now

**Review by sudermanjr — APPROVED** [review]
LGTM. Thanks for all the changes!

**Pull request merged**